File: teavm/src/main/java/net/sailware/bclicker/teavm/TeaVMBuilder.java
package net.sailware.bclicker.teavm;

import com.github.xpenatan.gdx.backends.teavm.config.AssetFileHandle;
import com.github.xpenatan.gdx.backends.teavm.config.TeaBuildConfiguration;
import com.github.xpenatan.gdx.backends.teavm.config.TeaBuilder;
import com.github.xpenatan.gdx.backends.teavm.config.plugins.TeaReflectionSupplier;
import com.github.xpenatan.gdx.backends.teavm.gen.SkipClass;
import java.io.File;
import java.io.IOException;
import org.teavm.tooling.TeaVMTool;
import org.teavm.vm.TeaVMOptimizationLevel;

/** Builds the TeaVM/HTML application. */
@SkipClass
public class TeaVMBuilder {
    public static void main(String[] args) throws IOException {
        TeaBuildConfiguration teaBuildConfiguration = new TeaBuildConfiguration();
        teaBuildConfiguration.assetsPath.add(new AssetFileHandle("../assets"));
        teaBuildConfiguration.webappPath = new File("build/dist").getCanonicalPath();

        // Register any extra classpath assets here:
        // teaBuildConfiguration.additionalAssetsClasspathFiles.add("net/sailware/bclicker/asset.extension");

        // Register any classes or packages that require reflection here:
        // TeaReflectionSupplier.addReflectionClass("net.sailware.bclicker.reflect");
        TeaReflectionSupplier.addReflectionClass("scala.runtime.LambdaDeserialize.bootstrap");

        TeaVMTool tool = TeaBuilder.config(teaBuildConfiguration);
        tool.setMainClass(TeaVMLauncher.class.getName());
        // For many (or most) applications, using the highest optimization won't add much to build time.
        // If your builds take too long, and runtime performance doesn't matter, you can change FULL to SIMPLE .
        tool.setOptimizationLevel(TeaVMOptimizationLevel.FULL);
        tool.setObfuscated(false);
        TeaBuilder.build(tool);
    }
}

File: teavm/src/main/scala/net/sailware/bclicker/teavm/TeaVMBuilder.scala
package net.sailware.bclicker.teavm

import com.github.xpenatan.gdx.backends.teavm.config.AssetFileHandle
import com.github.xpenatan.gdx.backends.teavm.config.TeaBuilder
import com.github.xpenatan.gdx.backends.teavm.config.TeaBuildConfiguration
import com.github.xpenatan.gdx.backends.teavm.config.plugins.TeaReflectionSupplier
import com.github.xpenatan.gdx.backends.teavm.gen.SkipClass
import java.io.File
import org.teavm.vm.TeaVMOptimizationLevel

@SkipClass
object TeaVMBuilder:

  def main(args: Array[String]): Unit =
    val buildConfig = TeaBuildConfiguration()
    buildConfig.assetsPath.add(AssetFileHandle("../assets"))
    buildConfig.webappPath = File("build/dist").getCanonicalPath()

    //TeaReflectionSupplier.addReflectionClass("")

    val tool = TeaBuilder.config(buildConfig)
    tool.setMainClass("net.sailware.bclicker.teavm.TeaVMLauncher")
    tool.setOptimizationLevel(TeaVMOptimizationLevel.FULL)
    tool.setObfuscated(false)
    TeaBuilder.build(tool)
